### ISSUE TITLE
Concurrency and randomness for with_items

### ISSUE THREAD

**Issue by acdha** [fcsonline/drill#83]
I was testing `drill` with a large CSV file of different item IDs. Is there any supported way to randomize the execution order? Right now this means that my concurrency is basically lock-step requesting the same items in the same order which maximizes the benefits from caching. It'd be really nice if there was a way to have the order be randomized other than, say, running multiple instances scripting `shuf` on a CSV file for each one.

**Comment by fcsonline** [comment]
It could be an interesting addition. We can add a `shuffle: true` option in all `with_items_*` directives in `drill`. What do you think?

**Comment by fcsonline** [comment]
Can you test this branch? https://github.com/fcsonline/drill/pull/84

**Comment by acdha** [comment]
On a tangent, I haven't looked at drill's internal data model much (or after #61) but there's one other area which took a little getting used to: concurrency and iterations are linked so there's no way to, say, run a large CSV file a single time with currency > 1 other than by using something like GNU Parallel to run multiple copies — (since drill is so simple to run that's not actually a bad option). 

It feels like it would be useful to have a way to expand the `with_items` directives so you could decouple those two settings and it'd be legal to, for example, say concurrency: 100 iterations:1 when you have a large input file so you'd still be able to things like the stats which wouldn't otherwise display if you terminated a large job before it finished many millions of requests.

**Comment by fcsonline** [comment]
This is an interesting topic regarding expanding all the `with_items` directives. I thought about it when I started the project but steps like your CSV example or this one would need much memory to be processed after been expanded:

```
  - name: Fetch some users by range
    request:
      url: /api/users/{{ item }}
    with_items_range:
      start: 1
      stop: 1000
```

Maybe it is something we can evaluate in the future if let us to do interesting things.

On the other hand, can I close this issue, now that we have the `shuffle` option, and handle this expanding topic in another issue?

**Comment by acdha** [comment]
Definitely - it’s doing what I needed and there’s a somewhat infinite number of things which can be added to a tool like this.

**Comment by fcsonline** [comment]
Great! Thanks!
